tests/js/session-tests.js

```diff
@@ -1033,6 +1034,113 @@ module.exports = {
         });
     },
 
+    async testClientResetAtOpen() {
+        // FIXME: try to enable for React Native
+        if (!isNodeProcess) {
+            return;
+        }
+        const fetch = require('node-fetch');
+
+        let called = false;
+        let user = await Realm.Sync.User.login('http://127.0.0.1:9080', Realm.Sync.Credentials.nickname('admin', true));
+        var realm;
+        const config = user.createConfiguration({ sync: { url: 'realm://127.0.0.1:9080/~/myrealm' } });
+        config.schema = [schemas.IntOnly];
+        config.sync.clientResyncMode = 'manual';
+        config.sync.fullSynchronization = true;
+        config.sync.error = (sender, error) => {
+            called = true;
+            try {
+                TestCase.assertEqual(error.name, 'ClientReset');
+                TestCase.assertDefined(error.config);
+                TestCase.assertNotEqual(error.config.path, '');
+                const path = realm.path;
+                realm.close();
+                Realm.Sync.initiateClientReset(path);
+                // open Realm with error.config, and copy required objects a Realm at `path`
+            }
+            catch (e) {
+            }
+        };
+
+        // open, download, create an object, upload and close
+        realm = await Realm.open(config);
+        await realm.syncSession.downloadAllServerChanges();
+        realm.write(() => {
+            realm.create(schemas.IntOnly.name, { intCol: 1 });
+        });
+        await realm.syncSession.uploadAllLocalChanges();
+        realm.close();
+
+        // delete Realm on server
+        let encodedPath = encodeURIComponent(`${user.identity}/myrealm`);
+        let url = new URL(`/realms/files/${encodedPath}`, user.server);
+        let options = {
+            headers: {
+                Authorization: `${user.token}`,
+                'Content-Type': 'application/json',
+            },
+            method: 'DELETE',
+        };
+        await fetch(url, options);
+
+        // open the Realm again and see it fail
+        return Realm.open(config).then(realm => {
+            Promise.reject();
+        }).catch(error => {
+            TestCase.assertTrue(called); // the error handler was called
+            TestCase.assertEqual(error.message, 'Synchronization no longer possible for client-side file');
+            Promise.resolve();
+        })
+    },
+
+    async testClientResyncDiscard() {
+        // FIXME: try to enable for React Native
+        if (!isNodeProcess) {
+            return;
+        }
+        const fetch = require('node-fetch');
```

Comment:
`nodeRequire`? Is there another fetch function (or another way we are already making requests) that we could reuse here?

Reply:
True. I have updated.